test/session_tests/DocumentQueryTest.ts

```diff
@@ -8,220 +8,289 @@ import {RequestExecutor} from "../../src/Http/Request/RequestExecutor";
+    await session.store<Universal>(new Universal('Universals_3', 'withNesting', 3, null, new Product(null, 'testing_order', 4)));
+    await session.store<Universal>(new Universal('Universals_4', 'withNesting', 4, null, new Product(null, 'testing_order', 4)));
+    await session.store<Universal>(new Universal('Universals_5', 'withNesting', 5, null, new Product(null, 'testing_order', 4)));
+    await session.store<Universal>(new Universal('Universals_6', 'withNesting', 6, null, new Product(null, 'testing_order', 4)));
+    await session.store<Universal>(new Universal('Universals_7', 'withNesting', 7, null, new Product(null, 'testing_order', 4)));
+    await session.store<Universal>(new Universal('Universals_8', 'withNesting', 8, null, new Product(null, 'testing_order', 4)));
+    await session.store<Universal>(new Universal('Universals_9', 'withNesting', 9, null, new Product(null, 'testing_order', 4)));
+    await session.store<Universal>(new Universal('Universals_10', 'withNesting', 10, null, new Product(null, 'testing_order', 4)));
+    await session.saveChanges();
   });
 
-  describe('Index checking', () => {
-    it('should query by dynamic index', async () => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product
-      }).whereEquals<string>('name', 'test101').get();
-        
-      expect(results[0].name).to.equals('test101');
-    });
-
-    it('should query by double index joined by "OR" operator', async () => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product
-      })
-      .whereEquals<string>('name', 'test101')
-      .whereEquals<number>('uid', 4).get();
-        
-      expect(results).to.have.lengthOf(2);
-    });
 
-    it('should query by double index joined by "AND" operator', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product,
-        waitForNonStaleResults: true,
-        usingDefaultOperator: QueryOperators.AND
-      })
-      .whereEquals<string>('name', 'test107')
-      .whereEquals<number>('uid', 5).get();
+  describe('Index checking', () => {
 
-      expect(results).to.have.lengthOf(1);
-    });
+    it('should query with nested objects', async() => {
+      const results: Universal[] = await store.openSession().query<Universal>({
+        indexName: 'Universals',
+        nestedObjectTypes: {product: Product}
+      }).whereEquals<string>('name', 'withNesting').get();
 
-    it('should query by whereIn', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product
-      }).whereIn<string>('name', ['test101', 'test107', 'test106']).get();
-      
-      expect(results).to.have.lengthOf(4);      
+      expect(results[0].product).to.be.instanceOf(Product);
```

Comment:
Assertions in these tests are useless. We just check we got something, but don't care what. Please add checks for the number of results you expect, the content of objects etc. This comment applies to every `it()` in this file.